trin-core/src/portalnet/metrics.rs

```diff
@@ -0,0 +1,57 @@
+use prometheus_exporter::{
+    self,
+    prometheus::{register_counter, Counter},
+};
+
+use crate::portalnet::types::messages::ProtocolId;
+
+/// General Metrics Strategy (wip)
+/// - Each module should maintain its own metrics reporter
+/// - When possible, use the lazy_static! approach
+/// - - https://romankudryashov.com/blog/2021/11/monitoring-rust-web-application/
+/// - - https://github.com/sigp/lighthouse/blob/c3a793fd73a3b11b130b82032904d39c952869e4/beacon_node/lighthouse_network/src/metrics.rs
+
+/// Overlay Service Metrics Reporter
+pub struct Metrics {
```

Comment:
If this relates to overlay metrics and it is not assumed implicitly (because `metrics.rs` file is in `portalnet` dir, should we rename this struct to `OverlayMetrics`?